shared_model/backend/protobuf/common_objects/asset.hpp

```diff
@@ -33,25 +33,22 @@ namespace shared_model {
       template <typename AssetType>
       explicit Asset(AssetType &&account)
           : CopyableProto(std::forward<AssetType>(account)),
-            assetId_(proto_->asset_id()),
-            domainId_(proto_->domain_id()),
-            precision_(proto_->precision()),
             blob_([this] { return BlobType(proto_->SerializeAsString()); }) {}
 
       Asset(const Asset &o) : Asset(o.proto_) {}
 
       Asset(Asset &&o) noexcept : Asset(std::move(o.proto_)) {}
 
       const interface::types::AssetIdType &assetId() const override {
-        return assetId_;
+        return proto_->asset_id();
       }
 
       const interface::types::DomainIdType &domainId() const override {
-        return domainId_;
+        return proto_->domain_id();
       }
 
-      const interface::types::PrecisionType &precision() const override {
-        return precision_;
+      interface::types::PrecisionType precision() const override {
```

Comment:
why remove const reference?

Reply:
It is just simple int

Reply:
Now yes, but in future we can potentially might change it with more complex object, can't we?
I thought that was the idea of using `using` for creating new types: so that in future we may change type in single place and don't worry about remaining code. Here it seems that if precision (for some unknown for me reason) would be changed with more complex type, redundant copying might happen